packages/react-vapor/src/components/listBox/ListBox.tsx

```diff
@@ -54,39 +59,45 @@ export class ListBox extends React.Component<IListBoxProps, {}> {
         const visibleLength = _.filter(this.props.items, (item: IItemBoxProps) => shouldShow(item) && !item.disabled)
             .length;
 
-        let index = 0;
+        let realIndex = 0;
         let activeSet = false;
         const items = _.chain(this.props.items)
             .filter(shouldShow)
             .map((item: IItemBoxProps) => {
                 let active = false;
+                const itemWithIndex: IITemBoxPropsWithIndex = {item: {...item}};

```

Comment:
Can you do this instead?

```suggestion
                const itemWithIndex: IITemBoxPropsWithIndex = {item};
```

Reply:
If I mutate the same item, it seems to have some side effects on some states holder. If I clone the object with a spread operator it seems to work fine.